tls/s2n_psk.c

```diff
@@ -328,7 +328,10 @@ int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struc
 
         /* s2n_decrypt_session_ticket appends a new PSK with the decrypted values. */
         POSIX_GUARD(s2n_decrypt_session_ticket(psk_list->conn));
+
+        POSIX_GUARD(s2n_stuffer_free(&psk_list->conn->client_ticket_to_decrypt));
```

Comment:
Do we need to call this in`s2n_connection_free_handshake` too? Or is it guaranteed to be freed by that point? If yes, then is it worth removing the `s2n_stuffer_wipe(&conn->client_ticket_to_decrypt)` call form `s2n_connection_wipe`?